examples/vision/cutmix.py

```diff
@@ -0,0 +1,391 @@
+
+    if (depth - 2) % 6 != 0:
+        raise ValueError("depth should be 6n+2 (eg 20, 32, 44 in [a])")
+    # Start the model definition
+    num_filters = 16
+    num_res_blocks = int((depth - 2) / 6)
+
+    inputs = keras.layers.Input(shape=input_shape)
+    x = resnet_layer(inputs=inputs)
+    # Instantiate the stack of residual units
+    for stack in range(3):
+        for res_block in range(num_res_blocks):
+            strides = 1
+            if stack > 0 and res_block == 0:  # The first layer but not the first stack
+                strides = 2  # Downsample
+            y = resnet_layer(inputs=x, num_filters=num_filters, strides=strides)
+            y = resnet_layer(inputs=y, num_filters=num_filters, activation=None)
+            if stack > 0 and res_block == 0:  # first layer but not first stack
+                # The linear projection residual shortcut connection to match
+                # the changed dimensions
+                x = resnet_layer(
+                    inputs=x,
+                    num_filters=num_filters,
+                    kernel_size=1,
+                    strides=strides,
+                    activation=None,
+                    batch_normalization=False,
+                )
+            x = keras.layers.add([x, y])
+            x = keras.layers.Activation("relu")(x)
+        num_filters *= 2
+
+    # Add the classifier on top of ResNet-20.
+    # (v1 does not use batch normalization after the last shortcut connection with ReLU)
+    x = keras.layers.AveragePooling2D(pool_size=8)(x)
+    y = keras.layers.Flatten()(x)
+    outputs = keras.layers.Dense(
+        num_classes, activation="softmax", kernel_initializer="he_normal"
+    )(y)
+
+    # Instantiate the model
+    model = keras.models.Model(inputs=inputs, outputs=outputs)
+    return model
+
+
+def training_model():
+    return resnet_v20((32, 32, 3), 20)
+
+
+initial_model = training_model()
+initial_model.save_weights("initial_weights.h5")
+
+"""
+## 1. Train the model with the dataset augmented by CutMix
+"""
+
+model = training_model()
+model.load_weights("initial_weights.h5")
+
+model.compile(loss="categorical_crossentropy", optimizer="adam", metrics=["accuracy"])
+model.fit(train_ds_cmu, validation_data=test_ds, epochs=5)
+
+test_loss, test_accuracy = model.evaluate(test_ds)
+print("Test accuracy: {:.2f}%".format(test_accuracy * 100))
+
+"""
+## 2. Train the model using the original non-augmented dataset
+"""
+
+model = training_model()
+model.load_weights("initial_weights.h5")
+model.compile(loss="categorical_crossentropy", optimizer="adam", metrics=["accuracy"])
+model.fit(train_ds_simple, validation_data=test_ds, epochs=5)
+
+test_loss, test_accuracy = model.evaluate(test_ds)
+print("Test accuracy: {:.2f}%".format(test_accuracy * 100))
+
+"""
+## Notes
+In this example, we trained our model for 15 epochs. In our experiment, the
+model with CutMix achieves a better accuracy on the CIFAR-10 dataset (80.02% in
```

Comment:
With how many epochs?

Reply:
Trying with MobileNetV2 model

Reply:
@fchollet I have tried with MobileNetV2 but not giving me desired result.